apis/kueue/webhooks/workload_webhook.go

```diff
@@ -159,23 +163,18 @@ func ValidateWorkloadUpdate(newObj, oldObj *kueue.Workload) field.ErrorList {
 	specPath := field.NewPath("spec")
 	allErrs = append(allErrs, ValidateWorkload(newObj)...)
 	allErrs = append(allErrs, apivalidation.ValidateImmutableField(newObj.Spec.PodSets, oldObj.Spec.PodSets, specPath.Child("podSets"))...)
-	allErrs = append(allErrs, validateQueueNameUpdate(newObj.Spec.QueueName, oldObj.Spec.QueueName, specPath.Child("queueName"))...)
+	if newObj.Spec.Admission != nil {
```

Comment:
It should be oldObj not newObj.

Reply:
When workload is admitted, I think the queueName wouldn't be empty. 

It's just a corner case. One workload is not admitted and the queueName is also empty, then the administrator manually set the admission and queueName, with current implementation, it will be forbidden.

Reply:
I changed it to be tied to whether the admission is set.

So at the time of admission you could still change the queueName, but not afterwards, unless the workloads is being preempted.